### DISCUSSION TITLE
Extensible Objects should be publicly accessible

### DISCUSSION THREAD

**Discussion by christianbraeunlich** [microsoft/alguidelines#169] [BC Best Practices]
When creating an extensible object
the __Access__ level should be set to __default__: _Public_

E.gg:

```al
codeunit 22020401 "Extensible CU"
{
    Extensible = true;
    
    trigger OnRun()
    begin
        Message('Happy Directions NA');
    end;
}
```

**Comment by christianbraeunlich** [comment]
Think I april-fooled my self on this one. The Default value of the `Access` property is `Public`. Discussion can be closed 🦦

**Comment by kine** [comment]
May be just use the negation - Objects, which should not be extended, should not be public... Or better - keep public only things which are really public and are creating "interface" of the app for other apps.

**Reply by christianbraeunlich** [reply]
Yes, you are absolutley right. The more I look at it, the more it scares me that the access modifier is public by default. I would have prefered Internal.

**Reply by waldo1001** [reply]
It's a bit double-sided, in all honesty.  I do agree, but we have been doing it "wrong" for years, and I'm very happy for it - as it's the only reason why we have been so efficient in implementing new customers with our own product 🤔.
Just to say - it's hard to predict what should be public ..
